### DISCUSSION TITLE
Routing based on presence of query parameters

### DISCUSSION THREAD

**Discussion by davidv1992** [tokio-rs/axum#2043] [Q&A] [answered]
I am building a webserver that needs to handle get requests to the same path very differently based on whether certain query parameters are present or not. Is it possible to do route these differently in axum, and if so, how?

**Comment by davidpdrsn** [comment]
Extract Query<HashMap<String, String>>. Then you check which keys are there. You can probably also use a struct with Option and #[serde(default)]

**Reply by davidv1992** [reply]
Hmm, then I really start doing a secondary layer of routing in my own code, which was not exactly what I was looking for. Was hoping to just do this as part of the main axum routing.

**Reply by davidpdrsn** [reply, marked as answer]
Yep you need to handle it yourself in the handle functions. Axum’s router doesn’t know about query params.

**Reply by davidv1992** [reply]
hmm, bit of a shame, but will have to deal with that then.
